src/Controls/src/Core/Handlers/Items/StructuredItemsViewHandler.Windows.cs

```diff
@@ -17,6 +19,32 @@ public partial class StructuredItemsViewHandler<TItemsView> : ItemsViewHandler<T
 
 		protected override IItemsLayout Layout { get => ItemsView?.ItemsLayout; }
 
+		protected override void ConnectHandler(ListViewBase platformView)
+		{
+			base.ConnectHandler(platformView);
+
+			if (Layout != null)
+				Layout.PropertyChanged += LayoutPropertyChanged;
+		}
+
+		protected override void DisconnectHandler(ListViewBase platformView)
+		{
+			base.DisconnectHandler(platformView);
+
+			if (Layout != null)
+				Layout.PropertyChanged -= LayoutPropertyChanged;
+		}
+
+		void LayoutPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (e.PropertyName == GridItemsLayout.SpanProperty.PropertyName)
+				UpdateItemsLayoutSpan();
+			else if (e.PropertyName == GridItemsLayout.HorizontalItemSpacingProperty.PropertyName || e.PropertyName == GridItemsLayout.VerticalItemSpacingProperty.PropertyName)
+				UpdateItemsLayoutItemSpacing();
+			else if (e.PropertyName == LinearItemsLayout.ItemSpacingProperty.PropertyName)
+				UpdateItemsLayoutItemSpacing();
```

Comment:
Would look a lot nicer with a switch expression and the removal of the else clauses.

Reply:
Agree with that. In this case, just continued following the if-else approach used so far in Xamarin. https://github.com/xamarin/Xamarin.Forms/blob/caab66bcf9614aca0c0805d560a34e176d196e17/Xamarin.Forms.Platform.Android/Renderers/ButtonRenderer.cs#L140